gensim/models/word2vec_inner.pyx

```diff
@@ -570,7 +573,11 @@ def train_batch_sg(model, sentences, alpha, _work, compute_loss):
             break  # TODO: log warning, tally overflow?
 
     # precompute "reduced window" offsets in a single randint() call
-    for i, item in enumerate(model.random.randint(0, c.window, effective_words)):
+    if reduced_windows:
+        window_size = model.random.randint(0, c.window, effective_words)
+    else:
+        window_size = [0] * effective_words
```

Comment:
Why `0`?

Reply:
Per above, the legacy summing logic means a `0` here leaves the full specified `window` in use. (You can essentially read the `reduced_windows` array as a list of values to 'reduce' the window by, with a 'reduction' of `0` leaving the full window in place.)

Reply:
Oh, OK. This variable is terribly named, then.

Reply:
But also, there's no need to create this list-of-0-values just to feed the next loop. The `if reduced_windows` branch should let each path fully initialize `c.reduced_windows`. EG:

```python
    if reduced_windows:
        for i, item in enumerate(model.random.randint(0, c.window, effective_words)):
            c.reduced_windows[i] = item
    else:
        c.reduced_windows[:] = 0
```

Reply:
That was indeed a bad variable name choice on my behalf; sorry.
The next commit will incorporate the suggested change.